app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php

```diff
@@ -0,0 +1,126 @@
+
+            $totalSpeciesAmount = $this->totalSpeciesAmount($country->slug);
+
+            $totalJobsCreated = $this->totalJobsCreated($country->slug);
+
+            $numberOfSites = $this->numberOfSites($country->slug);
+
+            $totalNurseries = $this->numberOfNurseries($country->slug);
+
+            $activeCountries[] = [
+                'country' => $country->label,
+                'number_of_projects' => $totalProjects,
+                'total_trees_planted' => $totalSpeciesAmount,
+                'total_jobs_created' => $totalJobsCreated,
+                'number_of_sites' => $numberOfSites,
+                'number_of_nurseries' => $totalNurseries,
+            ];
+        }
+
+        return $activeCountries;
+    }
+
+    public function projectsCountry($country)
+    {
+        return Project::where('framework_key', 'terrafund')
+            ->where('country', $country);
+    }
+
+    public function numberOfProjects($country)
+    {
+        $totalNumberOfProjects = $this->projectsCountry($country)
+                ->count();
+
+        return $totalNumberOfProjects;
+    }
+
+    public function totalSpeciesAmount($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
+        $totalSpeciesAmount = 0;
+        foreach ($projects as $project) {
+            $sites = $project->sites()->pluck('id')->toArray();
+            foreach ($sites as $siteId) {
+                $latestSiteReportId = SiteReport::where('site_id', $siteId)
+                    ->orderByDesc('due_at')
+                    ->value('id');
+                if ($latestSiteReportId !== null) {
+                    $totalSpeciesAmount += TreeSpecies::where('speciesable_id', $latestSiteReportId)->sum('amount');
+                }
+            }
+        }
+
+        return $totalSpeciesAmount;
+    }
+
+    public function totalJobsCreated($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
+        $totalJobs = 0;
+        foreach ($projects as $project) {
+            $latestProjectReport = ProjectReport::where('project_id', $project->id)
+                ->orderByDesc('due_at')
+                ->value(DB::raw('pt_total + ft_total'));
+            $totalJobs += $latestProjectReport;
+        }
+
+        return $totalJobs;
+    }
+
+    public function numberOfSites($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
+        $totalSites = 0;
+        foreach ($projects as $project) {
+            $totalSites += $project->sites()->count();
+        }
+
+        return $totalSites;
+    }
+
+    public function numberOfNurseries($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
+        $totalNurseries = 0;
+        foreach ($projects as $project) {
+            $totalNurseries += $project->nurseries()->count();
+        }
+
+        return $totalNurseries;
```

Comment:
I just noted that you are using the `foreach` method only for the sake of summing things.

Can you take a look at [documentation](https://laravel.com/docs/10.x/collections#method-sum) and see what is the best way to use in the endpoints?

Reply:
I have been looking the documentation and the best option is to use the SUM method, now i upload the changes.